### PR TITLE
Add `Display` and more `From<T>` implementations for `StrBytes`

### PR DESCRIPTION
**Pull request by Hackzzila**

### CODE REVIEW

**Comment by Hackzzila** [comment]
Looking back at this, I added `#[inline]` to Display because that is how the std lib does it. But should I add `#[inline]` for the other `From<T>` impls (which is what the stdlib does), or get rid of them all?

**Comment by tychedelia** [comment]
> Looking back at this, I added `#[inline]` to Display because that is how the std lib does it. But should I add `#[inline]` for the other `From<T>` impls (which is what the stdlib does), or get rid of them all?

I'd remove them, please. The `#[inline]` attribute is just a suggestion to the compiler and in practice these kinds of impls are almost certainly inlined regardless of its presence (or if they aren't, doesn't matter).

**Comment by Hackzzila** [comment]
Removed 👍

File: src/protocol/mod.rs [hunk 1]
```diff
@@ -78,6 +96,12 @@ mod str_bytes {
         }
     }
 
+    impl Display for StrBytes {
+        fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+            std::fmt::Display::fmt(&**self, f)
```
Comment (rukai) [inline src/protocol/mod.rs:101]: I was a little confused what this was doing.
Eventually I figured out its using the Deref impl to call `.as_str()`.

So to make it a little more obvious maybe call `.as_str()` directly?
`std::fmt::Display::fmt(self.as_str(), f)`

Also, we dont have any unit tests for StrBytes, and I think just adding one would go a long way in ensuring basic functionality works:
```rust
#[test]
fn strbytes_display() {
    let string = StrBytes::from_static_str("foobar");
    assert_eq!(format!("{string}"), "foobar".to_owned());
}
```

But these are both minor things, I'm happy to approve now.

**Review by rukai — APPROVED** [review]
This PR will make `StrBytes` far more convenient to use, thanks!

**Review by tychedelia — APPROVED** [review]

**Pull request merged**